### ISSUE TITLE
Building with Xcode 10

### ISSUE THREAD

**Issue by rajivshah3** [janeasystems/nodejs-mobile#123]
* **Project**: `nodejs-mobile`
* **Version**: latest from `mobile-master`
* **Mobile device**: N/A
* **Mobile OS and version**: iOS 12
* **Development Node.js**: `v8.11.4`
* **Development platform**: macOS Mojave 10.14 Beta (18A365a) - Xcode Version 10.0 (10A255)

I'm trying to build the iOS framework with `./tools/ios_framework_prepare.sh` but it keeps failing after a while:
```
Printing command line options back to the user:
_CXX=
_CC=
BUILD_TYPE=Release
MULTICORE_BUILD=
ICU_PATH=-DNO_ICU_PATH_GIVEN_SH=1
CMAKE_GEN=
MAKE=make V=1

ERROR: Minimum required Clang version is 3.7
  ln -f "../deps/chakrashim/core/lib/Jsrt/ChakraCore.h" "../deps/chakrashim/include/ChakraCore.h" 2>/dev/null || (rm -rf "../deps/chakrashim/include/ChakraCore.h" && cp -af "../deps/chakrashim/core/lib/Jsrt/ChakraCore.h" "../deps/chakrashim/include/ChakraCore.h")
make[1]: *** [/Users/rajiv/nodejs-mobile/out/Release/../../deps/chakrashim/core/out/Release/lib/libChakraCoreStatic.a] Error 1
make[1]: *** Waiting for unfinished jobs....
make: *** [node] Error 2
```
I just updated to Xcode 10. I used to be able to compile successfully with Xcode 9. For some reason, it thinks my Clang version is below 3.7, even though this is the output of `clang -v`:
```
Apple LLVM version 10.0.0 (clang-1000.11.45.2)
Target: x86_64-apple-darwin18.0.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
Found CUDA installation: /usr/local/cuda, version 8.0
```

Do I need to change any settings for Xcode 10?

**Comment by jaimecbernardo** [comment]
Hi @rajivshah3,

CMake detection used to expect a single digit on the major version number.
Applying the fix from upstream ChakraCore to `deps/chakrashim/core/build.sh` should fix it for Xcode 10:
https://github.com/Microsoft/ChakraCore/commit/6ed4bab072a3a97db8a2a3cbe0bcbc7efd904608

Please let us know if this change fixes your issue.

**Comment by rajivshah3** [comment]
Thanks so much @jaimecbernardo ! It works now. I fixed this in #125
